conda-store-server/conda_store_server/server/templates/create.html

```diff
@@ -26,6 +26,8 @@
         <label for="specification">Specification:</label>
         <div id="editor" style="height: 400px;">{{ specification or "" }}</div>
         <textarea name="specification" id="specification" style="display: none;"></textarea>
```

Comment:
I don't like how we are using the same textarea for two different things:

- When the input is `environment.yml`, we have all the information we need. This is good.
- When the input is a conda-lock, we need `name` and `description`. Since these fields are not part of the conda-lock schema, we embed the lockfile in a `lockfile` key. This is weird for the user because they will want to assume this collects the contents of the file. But it's not, they will be forced to edit after copy pasting.

The best solution IMO would be to revamp the UI a bit:

- As mentioned before, place a dropdown _above_ the text area so users can choose the type of input. The dropdown will have some events attached:
  - If the input file is a regular environment.yml, present the normal textarea
  - If the input file is a lockfile, then add two input fields with `name` and `description`. The content of the lockfile will be added to the textarea. For the backend, these three fields can be merged together in the data model you created (name, description and lockfile), but for the user shouldn't have to carry that abstraction burden.

That said, is `description` part of the `environment.yml` specification, or are we just adding it as a free form key because conda doesn't complain? Because the cheap way out of this for now would be, if conda-lock allows that, to add `name` and `description` top-level keys, and some sort of validation. But I don't like that hack.